packages/service-library/src/servicelib/base_distributed_identifier.py

```diff
@@ -66,10 +70,14 @@ def __init__(
         self._cleanup_task: Task | None = None
 
     async def setup(self) -> None:
+        lock_key = f"lock:{self.class_path()}:base-distributed-identifier"
+        lock_value = json.dumps({})
         self._cleanup_task = start_periodic_task(
-            self._cleanup_unused_identifiers,
+            exclusive(self._redis_client_sdk, lock_key=lock_key, lock_value=lock_value)(
```

Comment:
regarding your comment in the mattermost chat. I think you misunderstood the idea of using exclusive here. this effectively means that _cleanup_unused_identifiers will be run by only 1 process at a time. not sure what you were looking for here.